game/src/components/navbar.tsx

```diff
@@ -5,42 +5,59 @@ import { LogOut } from "lucide-react";
 import Link from "next/link";
 import { Button } from "./ui/button";
+import { useEffect, useState } from "react";
+import { ethers } from "ethers";
 
 export const Navbar = () => {
-  const { user, logout } = usePrivy();
-  const walletAddress = (user?.linkedAccounts as WalletWithMetadata[])?.find(
-    (a) => a.connectorType !== "embedded"
-  )?.address;
+    const { user, logout } = usePrivy();
+    const walletAddress = (user?.linkedAccounts as WalletWithMetadata[])?.find(
+        (a) => a.connectorType !== "embedded"
+    )?.address;
 
-  // const { resolvedTheme } = useTheme();
-  // const theme = resolvedTheme === "dark" ? "dark" : "light";
+    const [ensName, setEnsName] = useState<string | null>(null);
 
-  return (
-    <div className="flex justify-between flex-wrap p-6 px-4 m-auto w-full lg:w-[1500px]">
-      <Link
-        href={"/"}
-        className="text-4xl font-bold select-none cursor-pointer"
-      >
-        <Image
-          src={"https://chess.stf.xyz/logo-white.svg"}
-          width={150}
-          height={150}
-          alt="dchess-logo"
-        />
-      </Link>
-      <div className="flex gap-4 place-items-center">
-        {!!walletAddress && (
-          <div className="flex gap-4 items-center">
-            <p className="font-mono">
-              <b>AA Wallet:</b> {formatHash(user?.wallet?.address || "...")}
-            </p>
-            <Button onClick={logout} variant="outline" size="icon">
-              <LogOut />
-            </Button>
-          </div>
-        )}
-        {/* <ThemeToggle /> */}
-      </div>
-    </div>
-  );
-};
+    const mainnetProvider = new ethers.providers.JsonRpcProvider(process.env.NEXT_PUBLIC_ETHEREUM_RPC);
+
+    useEffect(() => {
+        const resolveEnsName = async () => {
+            if (walletAddress) {
+                try {
+                    const resolvedName = await mainnetProvider.lookupAddress(walletAddress);
+                    setEnsName(resolvedName || formatHash(walletAddress));
+                } catch (error) {
+                    setEnsName(formatHash(walletAddress));
+                }
+            }
+        };
+        resolveEnsName();
+    }, [walletAddress]);
+
+    return (
+        <div className="flex justify-between flex-wrap p-6 px-4 m-auto w-full lg:w-[1500px]">
+            <Link
+                href={"/"}
+                className="text-4xl font-bold select-none cursor-pointer"
+            >
+                <Image
+                    src={"https://chess.stf.xyz/logo-white.svg"}
+                    width={150}
+                    height={150}
+                    alt="dchess-logo"
+                />
+            </Link>
+            <div className="flex gap-4 place-items-center">
+                {!!walletAddress && (
+                    <div className="flex gap-4 items-center">
+                        <p className="font-mono">
+                            <b>AA Wallet:</b> {ensName || formatHash(walletAddress)}
```

Comment:
Few changes here, the `walletAddress` will be the user's non-custodial wallet address.
So you might have to show it like [here](https://github.com/stackrlabs/hackerpack/blob/c2514490281d5bc84740597410db872db335d1c2/web/src/components/navbar.tsx#L31)

Since this wallet is `Injected` and the one with `connectorType` as `embedded` is `AA Wallet`.